tests/ImageSharp.Tests/Formats/Jpg/JpegColorConverterTests.cs

```diff
@@ -40,323 +37,273 @@ public JpegColorConverterTests(ITestOutputHelper output)
+        public void FromGrayscaleBasic(int seed) =>
+            this.TestConverter(new JpegColorConverterBase.FromGrayscaleScalar(8), 1, seed);
 
         [Theory]
-        [MemberData(nameof(CommonConversionData))]
-        public void FromRgbVector8(int inputBufferLength, int resultBufferLength, int seed)
+        [MemberData(nameof(Seeds))]
+        public void FromGrayscaleVector(int seed)
         {
-            if (!SimdUtils.HasVector8)
+            var converter = new JpegColorConverterBase.FromGrayScaleVector(8);
+
+            if (!converter.IsAvailable)
             {
-                this.Output.WriteLine("No AVX2 present, skipping test!");
+                this.Output.WriteLine(
+                    $"Skipping test - {converter.GetType().Name} is not supported on current hardware.");
                 return;
             }
 
-            ValidateConversion(
-                new JpegColorConverter.FromRgbVector8(8),
-                3,
-                inputBufferLength,
-                resultBufferLength,
-                seed);
+            FeatureTestRunner.RunWithHwIntrinsicsFeature(
+                RunTest,
+                seed,
+                HwIntrinsics.AllowAll | HwIntrinsics.DisableAVX);
+
+            static void RunTest(string arg) =>
+                ValidateConversion(
+                    new JpegColorConverterBase.FromGrayScaleVector(8),
+                    1,
+                    FeatureTestRunner.Deserialize<int>(arg));
         }
 
+#if SUPPORTS_RUNTIME_INTRINSICS
+        [Theory]
+        [MemberData(nameof(Seeds))]
+        public void FromGrayscaleAvx2(int seed) =>
+            this.TestConverter(new JpegColorConverterBase.FromGrayscaleAvx(8), 1, seed);
+#endif
+
         [Theory]
-        [MemberData(nameof(CommonConversionData))]
-        public void FromRgbAvx2(int inputBufferLength, int resultBufferLength, int seed)
+        [MemberData(nameof(Seeds))]
+        public void FromRgbBasic(int seed) =>
+            this.TestConverter(new JpegColorConverterBase.FromRgbScalar(8), 3, seed);
+
+        [Theory]
+        [MemberData(nameof(Seeds))]
+        public void FromRgbVector(int seed)
         {
-            if (!SimdUtils.HasAvx2)
+            var converter = new JpegColorConverterBase.FromRgbVector(8);
+
+            if (!converter.IsAvailable)
             {
-                this.Output.WriteLine("No AVX2 present, skipping test!");
+                this.Output.WriteLine(
+                    $"Skipping test - {converter.GetType().Name} is not supported on current hardware.");
                 return;
             }
 
-            ValidateConversion(
-                new JpegColorConverter.FromRgbAvx2(8),
-                3,
-                inputBufferLength,
-                resultBufferLength,
-                seed);
+            FeatureTestRunner.RunWithHwIntrinsicsFeature(
+                RunTest,
+                seed,
+                HwIntrinsics.AllowAll | HwIntrinsics.DisableAVX);
```

Comment:
> It doesn't affect coverage afaik.

Makes sense, 3.1+ builds should cover the SSE cases, so it's fine if 2.1 and Framework are being executed against AVX only. Maybe you can do something like:

```C#
#if SUPPORTS_RUNTIME_INTRINSICS
            HwIntrinsics intrinsicsConfig =  HwIntrinsics.AllowAll | HwIntrinsics.DisableAVX;
#else
            HwIntrinsics intrinsicsConfig = HwIntrinsics.AllowAll;
#endif

            FeatureTestRunner.RunWithHwIntrinsicsFeature(
                RunTest,
                seed,
                intrinsicsConfig);

```